### PR TITLE
pvc controller sync unbound pvc and requeue after bound failed

### PR DESCRIPTION
**Pull request by goushicui**
#### What type of PR is this?
/kind bug

#### Which issue(s) this PR fixes:

Fixes #118225

#### Special notes for your reviewer:
/assign @jsafrane 

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by Tusenka — APPROVED** [review]

**Comment by goushicui** [comment]
@jsafrane

File: pkg/controller/volume/persistentvolume/pv_controller_base.go [hunk 1]
```diff
@@ -574,7 +575,10 @@ func (ctrl *PersistentVolumeController) claimWorker(ctx context.Context) {
 		if err == nil {
 			// The claim still exists in informer cache, the event must have
 			// been add/update/sync
-			ctrl.updateClaim(ctx, claim)
+			if err := ctrl.updateClaim(ctx, claim); err != nil || claim.Status.Phase == v1.ClaimPending {
+				logger.V(2).Info("Error update claim status", "claimKey", key, "err", err)
+				ctrl.claimQueue.AddAfter(claimToClaimKey(claim), time.Second*1)
```
Comment (sttts) [inline pkg/controller/volume/persistentvolume/pv_controller_base.go:580]: why not rely on exponential backoff of the queue? I.e. `ctrl.claimQueue.AddRateLimited(key)`

Comment (goushicui) [reply pkg/controller/volume/persistentvolume/pv_controller_base.go]: @sttts  thx, updated

**Comment by jsafrane** [comment]
While I think this PR is not bad, I think it does not help with https://github.com/kubernetes/kubernetes/issues/118225.

If I read it correctly, this PR will add PVCs that failed `updateClaim` to  `claimQueue` _faster_ than in the next periodic sync (15s). But the problem in #118225 is that `claimQueue` is already too large to be processed in 15 seconds, adding items to it faster won't fix the issue.

**Comment by kkkkun** [comment]
> But the problem in #118225 is that `claimQueue` is already too large to be processed in 15 seconds, adding items to it faster won't fix the issue.

Yes, you are right. So we do as follow.

1. Firstly, when sync failed added to queue again which is this PR does. 
2. Secondly, adjustment parameters `pvclaimbinder-sync-period` to 30s.

**Comment by kkkkun** [comment]
> > But the problem in #118225 is that `claimQueue` is already too large to be processed in 15 seconds, adding items to it faster won't fix the issue.
> 
> Yes, you are right. So we do as follow.
> 
> 1. Firstly, when sync failed added to queue again which this PR does.
> 2. Secondly, adjustment parameters `pvclaimbinder-sync-period` to 30s.
> 
> In my test, pvc and pv have 50k. then create pvc which local-voulme-provision would create pv.

The PR is matched with  https://github.com/kubernetes/kubernetes/pull/118852#issuecomment-1638117837

**Comment by kkkkun** [comment]
> > But the problem in #118225 is that `claimQueue` is already too large to be processed in 15 seconds, adding items to it faster won't fix the issue.
> 
> Yes, you are right. So we do as follow.
> 
> 1. Firstly, when sync failed added to queue again which this PR does.
> 2. Secondly, adjustment parameters `pvclaimbinder-sync-period` to 30s.
> 
> In my test, pvc and pv have 50k. then create pvc which local-voulme-provision would create pv.

The PR is matched with https://github.com/kubernetes/kubernetes/issues/71761#issuecomment-445258408

**Pull request closed without merging**